src/openforms/forms/api/serializers/tests/test_action_serializers.py

```diff
@@ -0,0 +1,169 @@
+from django.utils.translation import ugettext_lazy as _
+
+from hypothesis import given
+from hypothesis.extra.django import TestCase as HypothesisTestCase
+from hypothesis_jsonschema import from_schema
```

Comment:
hypothesis-jsonschema is designed to work with json-schema, which is a (draft) standard for describing json schema's in JSON. It is used internally by OpenAPI speficitation, for example.

This does do what you think it does in terms of "throw whatever valid JSON at the serializer and see if it explodes", as json-schema is much more strict and structured.

So, please remove this dependency, and instead write the test using existing constructs we have:

1. We have a JSON search strategy here: https://github.com/open-formulieren/open-forms/blob/master/src/openforms/tests/search_strategies.py#L29
2. Example usage: https://github.com/open-formulieren/open-forms/blob/master/src/openforms/variables/tests/test_validators.py#L71

Then finally, strings are the only type that we do allow/expect, so we need to exclude those, you can use `filter` for that.

This should result in testcode roughly looking like:

```py
json_without_strings = json_values.filter(lambda c: not isinstance(c, str))

@given(json_without_strings)
def test_date_validation_arbitrary_json_input(self, json_value):
    serializer = LogicComponentActionSerializer(...)
    
    is_valid = serializer.is_valid()

    self.assertFalse(is_valid)
    self.assertEqual(
            serializer.errors["action"]["value"],
            _(
                "Value for date variable must be a string in the format "
                "yyyy-mm-dd (e.g. 2023-07-03)"
            ),
        )
```

Hypothesis will take care of generating numbers, booleans, objects, arrays... for that test and run it with a number of combinations. The end result of anything that is not a string should be an invalid serializer.

Reply:
If you don't need the strategy elsewhere you could also use [assume](https://hypothesis.readthedocs.io/en/latest/details.html#hypothesis.assume) in the single test that needs some cases filtered:
```py
@given(json_values)
def test_something(self, json_value):
    assume(not isinstance(json_value, str))
    ...  # assertions....
```

`assume` is like `assert` that doesn't fail the test, but "fails the generated value" so to speak.

Reply:
This test doesn't work. The validator on action values in `openforms/forms/api/validators.py` will throw an `AssertionError` for arrays, lists, etc.